src/dioptra/restapi/v1/workflows/service.py

```diff
@@ -65,3 +67,63 @@ def get(self, job_id: int, file_type: FileTypes, **kwargs) -> IO[bytes]:
             file_type=file_type,
             logger=log,
         )
+
+
+class SignatureAnalysisService(object):
+    """The service methods for performing signature analysis on a file."""
+
+    def post(self, fileContents: str, **kwargs) -> dict[str, List[dict[str, Any]]]:
+        """Perform signature analysis on a file.
+
+        Args:
+            filename: The name of the file.
+            file_contents: The contents of the file.
+
+        Returns:
+            A dictionary containing the signature analysis.
+        """
+        log: BoundLogger = kwargs.get("log", LOGGER.new())
+        log.debug(
+            "Performing signature analysis",
+            python_source=fileContents,
+        )
+
+        signatures = list(
+            get_plugin_signatures(
+                python_source=fileContents,
+            )
+        )
+
+        endpoint_analyses = []
+        for signature in signatures:
+            function_name = signature["name"]
+            function_inputs = signature["inputs"]
+            function_outputs = signature["outputs"]
+            inferences = signature["suggested_types"]
+            endpoint_analysis = {}
+
+            endpoint_analysis["name"] = function_name
+            endpoint_analysis["inputs"] = function_inputs
+            endpoint_analysis["outputs"] = function_outputs
+
+            # Compute the suggestions for the unknown types
+
+            missing_types = []
+
+            for inference in inferences:
+                suggested_type = inference[
+                    "suggestion"
+                ]  # replace this with resource id's for suggestions
+                original_annotation = inference[
+                    "type_annotation"
+                ]  # do a database lookup with this
+                missing_types += [
+                    {
+                        "description": original_annotation,
+                        "name": suggested_type,
+                    }
+                ]
+
+            endpoint_analysis["missing_types"] = missing_types
+            endpoint_analyses += [endpoint_analysis]
+        return {"plugins": endpoint_analyses}
```

Comment:
Note that `"plugins"` in the returned dictionary may need to be changed, see my comments on `workflows/schema.py`.